qiskit_aer/backends/aerbackend.py

```diff
@@ -345,6 +392,65 @@ def _run(self, qobj, job_id='', format_result=True):
             return self._format_results(output)
         return output
 
+    def _run_direct(self, circuits, noise_model, config, job_id='', format_result=True):
+        """Run a job"""
+        # Start timer
+        start = time.time()
+
+        # Take metadata from headers of experiments to work around JSON serialization error
+        metadata_list = []
+        for idx, circ in enumerate(circuits):
+            if hasattr(circ, "metadata") and circ.metadata:
+                metadata = circ.metadata
+                metadata_list.append(metadata)
+                circ.metadata = {}
+                if "id" in metadata:
+                    circ.metadata["id"] = metadata["id"]
+                circ.metadata["metadata_index"] = idx
```

Comment:
Where do these come from? I'm not sure I understand what is setting `id` and why we need to set the metadata index on the circuit object here.

Reply:
https://github.com/Qiskit/qiskit-aer/blob/ce9a8605545c671b653c07562d8283ca98fc951b/qiskit_aer/jobs/utils.py#L59-L88

`id` is set when "qobj" is divided for DASK execution. However, yes, here is not qobj execution and we can take it (until DASK supports circuit execution).